packages/balance-monitor/balance-monitor/balance_monitor.go

```diff
@@ -0,0 +1,223 @@
+package balanceMonitor
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/big"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/exp/slog"
+)
+
+type ethClient interface {
+	BalanceAt(ctx context.Context, account common.Address, blockNumber *big.Int) (*big.Int, error)
```

Comment:
The vast majority of these methods are unused, and dont need to be stubbed for the interface here. we only need BalanceAt and maybe some others that you actually use.

Reply:
Removed unused methods. The rest is needed (I just check them one by one by removing it)